MessageTerminal/MessageTerminal.cxx

```diff
@@ -0,0 +1,143 @@
+#include <unistd.h>
+
+#include <iostream>
+#include <sstream>
+
+#include "LinuxSerialDevice.h"
+#include "TestStandCommHost.h"
+
+#define BAUD_RATE 115200
+
+#define BASIC_CMD(_name)                    \
+void _name(istringstream& iss)              \
+{                                           \
+    if (comm._name()) cout << "OK" << endl; \
+    else cout << "ERROR" << endl;           \
+}
+
+using namespace std;
+
+LinuxSerialDevice device;
+TestStandCommHost comm(device);
+
+typedef void (*cmd_handler)(istringstream& iss);
+
+BASIC_CMD(ping);
+BASIC_CMD(get_status);
+BASIC_CMD(home);
+BASIC_CMD(stop);
+
+void move(istringstream& iss)
+{
+    uint16_t accel, hold_vel, dist;
+    AxisId axis;
+    Direction dir;
+
+    do {
+        // accel, hold_vel, dist
+        if (!iss.good()) break;
+        iss >> accel;
+        if (!iss.good()) break;
+        iss >> hold_vel;
+        if (!iss.good()) break;
+        iss >> dist;
+
+        string word;
+
+        // axis
+        if (!iss.good()) break;
+        iss >> word;
+        if (word == "x") axis = AXIS_X;
+        else if (word == "y") axis = AXIS_Y;
+        else break;
+
+        // dir
+        if (!iss.good()) break;
+        iss >> word;
+        if (word == "pos") dir = DIR_POSITIVE;
+        else if (word == "neg") dir = DIR_NEGATIVE;
+        else break;
+
+        if (comm.move(accel, hold_vel, dist, axis, dir)) cout << "OK" << endl;
+        else cout << "ERROR" << endl;
+
+        if (comm.recv_message(5000) && comm.received_message().id == MSG_ID_LOG) {
+            printf("%s\n", &(comm.received_message().data[1]));
```

Comment:
Is there a reason for mixing `cout` or `printf`? I also realize this is a test file so consistency/comments not a priority.

Reply:
I spent about 10 seconds trying to get `cout` to print formatted stuff and it didn't work so I just switched to `printf` whenever I needed to format the output.
I think maybe I'll just use `printf` everywhere.

Reply:
Leaving as is for now, can come back to this later